### ISSUE TITLE
[AIR] `TensorflowPredictor` incorrectly ravels predictions 

### ISSUE THREAD

**Issue by bveeramani** [ray-project/ray#25137]
### What happened + What you expected to happen

I used `BatchPredictor` to classify an image dataset. I expected the output dataset to contain logits, but instead it contains scalars.

I got 
```
>>> outputs.show(1)
{'predictions': 0.0}
```

but I should've got

```
{'predictions': array([0., 0., 0., 0., 0., 0., 0., 0., 0., 0.], dtype=float32)}
```

**Importat: you'll only run into this error if you've patched #25125 and #25124 first!**

### Versions / Dependencies

**Ray**: master
**OS**: MacOS
**Python**: 3.8.12

### Reproduction script

```
import ray
from ray.ml.predictors.integrations.tensorflow import TensorflowPredictor
import tensorflow as tf
from tensorflow.keras import models, layers
from ray.ml.batch_predictor import BatchPredictor
from ray.ml.checkpoint import Checkpoint

batch_size = 4
height = 32
width = 32
num_channels = 3
num_classes = 10


def build_model():
    model = models.Sequential()

    model.add(
        layers.Conv2D(
            6, (5, 5), activation="relu", input_shape=(height, width, num_channels)
        )
    )
    model.add(layers.MaxPooling2D((2, 2)))
    model.add(layers.Conv2D(16, (5, 5), activation="relu"))
    model.add(layers.MaxPooling2D((2, 2)))
    model.add(layers.Flatten())
    model.add(layers.Dense(120, activation="relu"))
    model.add(layers.Dense(84, activation="relu"))
    model.add(layers.Dense(num_classes))
    return model


# Mocking a checkpoint so I can use `BatchPredictor`.
model = build_model()
model.build(input_shape=(0, 32, 32, 3))
checkpoint = Checkpoint.from_dict({"model": model.get_weights()})

batch_predictor = BatchPredictor.from_checkpoint(
    checkpoint=checkpoint,
    predictor_cls=TensorflowPredictor,
    model_definition=build_model,
)

dataset = ray.data.range_tensor(batch_size, shape=(height, width, num_channels))
outputs = batch_predictor.predict(dataset)
outputs.show(1)
```

### Issue Severity

High: It blocks me from completing my task.
